From b410c3db5fe12f83715dde7abb892a85b4310913 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Thu, 19 Sep 2019 17:05:45 +0200
Subject: mingw: use MSYS2 quoting even when spawning shell scripts

At the point where `mingw_spawn_fd()` is called, we already have a full
path to the script interpreter in that scenario, and we pass it in as
the executable to run, while the `argv` reflect what the script should
receive as command-line.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
(cherry picked from commit 49f7a76d57c78e55f2f72e44664824f3fecb8837)
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 compat/mingw.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/compat/mingw.c b/compat/mingw.c
index d0add1b18c..abc4723487 100644
--- a/compat/mingw.c
+++ b/compat/mingw.c
@@ -1443,7 +1443,8 @@ static pid_t mingw_spawnve_fd(const char *cmd, const char **argv, char **deltaen
 	BOOL ret;
 	HANDLE cons;
 	const char *(*quote_arg)(const char *arg) =
-		is_msys2_sh(*argv) ? quote_arg_msys2 : quote_arg_msvc;
+		is_msys2_sh(cmd ? cmd : *argv) ?
+		quote_arg_msys2 : quote_arg_msvc;
 
 	do_unset_environment_variables();
 
-- 
2.24.0.393.g34dc348eaf

